ouroboros-network-framework/src/Ouroboros/Network/InboundGovernor/ControlChannel.hs

```diff
@@ -53,31 +55,41 @@ instance Show peerAddr
 
 
 
--- | A Server control channel which instantiates 'handle'.
+-- | A Server control channel which instantiates to 'NewConnection' and 'Handle'.
+--
+-- It allows to pass 'STM' transactions which will resolve to 'NewConnection'.
+-- Server's monitoring thread is the consumer of these messages; there are two
+-- producers: accept loop and connection handler for outbound connections.
 --
 type ServerControlChannel (muxMode :: MuxMode) peerAddr versionData bytes m a b =
-    ControlChannel peerAddr (Handle muxMode peerAddr versionData bytes m a b) m
+    ControlChannel (NewConnection peerAddr (Handle muxMode peerAddr versionData bytes m a b)) m
+
+-- | Control Channel between Server and Outbound Governor.
+--
+-- Control channel that is meant to share inbound connections with the Peer
+-- Selection Governor. So the consumer is the Governor and Producer is the
+-- Server.
+--
+type GovernorControlChannel peerAddr m =
+    ControlChannel (peerAddr, PeerSharing) m
```

Comment:
If we want to reuse this type in the `outbound-governor` we also should rename this module; otherwise it's confusing to use something below `Ouroboros.Network.InboundGovernor` in the `outbound-governor`.

Since both channels are used by the `connection-manager`, maybe we should rename this module to `Ouroboros.Network.ConnectionManager.ControlChannel`.

If we are at this maybe there's actually a better name name `ControlChannel`, it's not much about controlling something but passing information / notifications around.

Reply:
The `Ouroboros.Network.InboundGovernor.OutboundGovernorInfoChannel` should replace this type alias here.